src/metaswitch/clearwater/queue_manager/test/test_remove_from_queue_success.py

```diff
@@ -0,0 +1,88 @@
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import unittest
+from .test_base import BaseQueueTest
+from metaswitch.clearwater.etcd_shared.test.mock_python_etcd import EtcdFactory
+from metaswitch.clearwater.queue_manager.etcd_synchronizer import EtcdSynchronizer
+from .plugin import TestPlugin
+from mock import patch, MagicMock
+from threading import Thread
+from time import sleep
+import json
+
+alarms_patch = patch("metaswitch.clearwater.queue_manager.alarms.issue_alarm", new=MagicMock)
+
+class RemoveFromQueueSuccessTest(BaseQueueTest):
+    @patch("etcd.Client", new=EtcdFactory)
+    def setUp(self):
+        alarms_patch.start()
+        self._p = TestPlugin()
+        self._e = EtcdSynchronizer(self._p, "10.0.0.1", "local", "clearwater", "node")
+        self._e.WAIT_FOR_TIMER_POP = 0
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_remove_from_queue_success(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": [{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"PROCESSING\"}]}")
+        self._e.remove_from_queue(True)
+
+        val = json.loads(self._e._client.read("/clearwater/local/configuration/queue_test").value)
+        self.assertEqual(0, len(val.get("ERRORED")))
+        self.assertEqual(1, len(val.get("COMPLETED")))
+        self.assertEqual(0, len(val.get("QUEUED")))
+        self.assertEqual("10.0.0.1-node", val.get("COMPLETED")[0]["ID"])
+        self.assertEqual("DONE", val.get("COMPLETED")[0]["STATUS"])
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_remove_from_queue_success_still_in_queue(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": [{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"PROCESSING\"},{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"QUEUED\"}]}")
+        self._e.remove_from_queue(True)
+
+        val = json.loads(self._e._client.read("/clearwater/local/configuration/queue_test").value)
+        self.assertEqual(0, len(val.get("ERRORED")))
+        self.assertEqual(0, len(val.get("COMPLETED")))
+        self.assertEqual(1, len(val.get("QUEUED")))
+        self.assertEqual("10.0.0.1-node", val.get("QUEUED")[0]["ID"])
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_remove_from_queue_success_not_front_of_queue(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"UNRESPONSIVE\"}], \"COMPLETED\": [], \"QUEUED\": [{\"ID\":\"10.0.0.2-node\",\"STATUS\":\"PROCESSING\"}]}")
+        self._e.remove_from_queue(True)
```

Comment:
I don't really understand this test - it looks like it's checking that `remove_from_queue` doesn't do anything?